### PR TITLE
Adds automated support to test `Init node cannot be deleted` issue

### PR DESCRIPTION
**Pull request by Josh-Diamond**
adds a test and extensions to delete the init node of a downstream cluster, then verify successful deletion

Issue:
https://github.com/rancher/rancher/issues/42709

### CODE REVIEW

File: tests/framework/extensions/nodes/nodes.go
Comment (susesgartner) [inline tests/framework/extensions/nodes/nodes.go:36]: Style nit, Indent the lines after the if.

File: tests/framework/extensions/nodes/nodes.go
Comment (susesgartner) [inline tests/framework/extensions/nodes/nodes.go:75]: I would highly suggest replacing client.Management.Node.ByID with its Steve equivalent.

File: tests/framework/extensions/nodes/nodes.go
Comment (susesgartner) [inline tests/framework/extensions/nodes/nodes.go:4]: If possible please eliminate all things norman.

**Review by susesgartner — COMMENTED** [review]
There are a few style related nits but good work overall!

File: tests/framework/extensions/nodes/nodes.go
Comment (markusewalker) [inline tests/framework/extensions/nodes/nodes.go:1]: Just a reminder that as this is an extension, this needs to go to `rancher/shepherd`. So, whenever we're given the green light, this needs to stripped from this PR and put in a separate PR in that repo.

File: tests/framework/extensions/nodes/nodes.go
Comment (markusewalker) [inline tests/framework/extensions/nodes/nodes.go:23]: nit; in the function descriptions, can you have it say `GetInitNode accepts a client...`? Our godocs typically stylize functions in this way.

This same comment goes for all of the functions.

File: tests/framework/extensions/nodes/nodes.go
Comment (markusewalker) [inline tests/framework/extensions/nodes/nodes.go:26]: nit; can you remove the space?

File: tests/framework/extensions/nodes/nodes.go
Comment (markusewalker) [inline tests/framework/extensions/nodes/nodes.go:55]: nit; remove the line here.

File: tests/framework/extensions/nodes/nodes.go
Comment (markusewalker) [inline tests/framework/extensions/nodes/nodes.go:57]: nit; remove the line here.

File: tests/framework/extensions/nodes/nodes.go
Comment (markusewalker) [inline tests/framework/extensions/nodes/nodes.go:76]: nit; remove the line here.

File: tests/framework/extensions/nodes/nodes.go
Comment (markusewalker) [inline tests/framework/extensions/nodes/nodes.go:84]: Same nit with the function description.

File: tests/v2/validation/deleting/delete_init_node_test.go
Comment (markusewalker) [inline tests/v2/validation/deleting/delete_init_node_test.go:35]: Let's talk offline about this design. I think we should follow the practice we're doing in the other deleting test/scaling tests where we are just grabbing the cluster ID and then proceeding to the functionality you have.

In other words, I think you should have a `delete_init_node.go` file where everything after `GetClusterIDByName` is referenced; the `TestDeleteInitNode` should call one function that lives in `delete_init_node.go` that will then run every other function you have here.

File: tests/framework/extensions/nodes/nodes.go
Comment (markusewalker) [inline tests/framework/extensions/nodes/nodes.go:117]: Not sure about having a `fatalf` here. I think having just erroring out should suffice.

File: tests/framework/extensions/nodes/nodes.go
Comment (markusewalker) [inline tests/framework/extensions/nodes/nodes.go:108]: nit; remove the line here.

File: tests/framework/extensions/nodes/nodes.go
Comment (markusewalker) [inline tests/framework/extensions/nodes/nodes.go:95]: nit; remove the line here.

File: tests/framework/extensions/nodes/nodes.go
Comment (markusewalker) [inline tests/framework/extensions/nodes/nodes.go:88]: nit; remove the line here.

File: tests/framework/extensions/nodes/nodes.go
Comment (Josh-Diamond) [reply tests/framework/extensions/nodes/nodes.go:55]: which line?

53? or 55? 🤔

File: tests/framework/extensions/nodes/nodes.go
Comment (Josh-Diamond) [reply tests/framework/extensions/nodes/nodes.go:57]: line 57? 🤔

File: tests/framework/extensions/nodes/nodes.go
Comment (Josh-Diamond) [reply tests/framework/extensions/nodes/nodes.go:117]: Maybe we can sync up on this?

Context:
returning `fmt.Error()` returned the error and exited, which was desired behavior.  When attempting to refactor to use `logrus` instead of `fmt`, my linter would freak out if i tried to return `logrus.Error()` -- it only wanted to be declared, not returned 🤔 so, removing the return and just declaring the logrus functions, the following occurred:

When using `logrus.Error()`, the error was logged, but not recognized as such by the test.  Ultimately the test proceeds.  This was not the case with using `logrus.Fatalf()`, which logged the error and immediately exited, which was the desired behavior I was looking for.

If there is a more ideal solution, I'm open to implementing it, but based on what I've tried, this gave the desired behavior I was looking for.

File: tests/v2/validation/deleting/delete_init_node.go
Comment (markusewalker) [inline tests/v2/validation/deleting/delete_init_node.go:20]: This can be removed since the proceeding functions already give verbose logging output.

File: tests/v2/validation/deleting/delete_init_node.go
Comment (markusewalker) [inline tests/v2/validation/deleting/delete_init_node.go:22]: nit - can you leave a comment as to why there is a need to have this called twice? We've talked offline of the need, but it will be good to have this documented.

**Review by markusewalker — COMMENTED** [review]
Overall, LGTM. Will wait for offline freeform jobs before approving.

File: go.mod
Comment (markusewalker) [inline go.mod:19]: Reminder that this is pointing to your forked repo. Once testing is done, please point to the official repo and not your forked repo.

File: go.sum
Comment (markusewalker) [inline go.sum:1333]: This will be addressed once you run `go mod tidy` after you have the official repo back in place.

**Review by markusewalker — COMMENTED** [review]
Left some nits, but overall looks good. I will approve once the shepherd PR gets merged. Please remember to open backport PRs once shepherd is merged and there are no more reviews for this PR.

**Review by markusewalker — APPROVED** [review]
LGTM

File: tests/v2/validation/deleting/delete_init_machine_test.go
Comment (slickwarren) [inline tests/v2/validation/deleting/delete_init_machine_test.go:38]: since this says `machine` I assume that's just for k3s/rke2. But can we make that more explicit? Following some of the new code added to have better naming in Qase, something like this 
```
		clusterID, err := clusters.GetV1ProvisioningClusterByName(s.client, s.client.RancherConfig.ClusterName)
		require.NoError(s.T(), err)

		cluster, err := tt.client.Steve.SteveType(ProvisioningSteveResourceType).ByID(clusterID)
		require.NoError(s.T(), err)

		updatedCluster := new(apisV1.Cluster)
		err = v1.ConvertToK8sType(cluster, &updatedCluster)
		require.NoError(s.T(), err)

		if strings.Contains(updatedCluster.Spec.KubernetesVersion, "rke2") {
			tt.name = "Replacing RKE2 " + tt.name
		} else {
			tt.name = "Replacing K3S " + tt.name
		}

		s.Run(tt.name, func() {
			deleteInitMachine(d.T(), d.client, clusterID)
	}
```

Comment (Josh-Diamond) [reply tests/v2/validation/deleting/delete_init_machine_test.go:38]: this is just for rke2/k3s 🙂

Comment (slickwarren) [reply tests/v2/validation/deleting/delete_init_machine_test.go:38]: still, I think we need to include the cluster type in the test name. Can you update the test function to do that, similar to the snippet above? this helps with Qase reporting.

**Pull request closed without merging**
